tests/std/tests/P0355R7_calendars_and_time_zones_time_point_and_durations/test.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <chrono>
+#include <limits>
+#include <type_traits>
+
+using namespace std;
+using namespace std::chrono;
+
+#define DURATION_TEST(TYPE, BITS, ...)                                                                                \
+    static_assert(                                                                                                    \
+        is_integral_v<TYPE::rep> && is_signed_v<TYPE::rep>, "chrono::" #TYPE "::rep is not a signed integral type."); \
+    static_assert(                                                                                                    \
+        numeric_limits<TYPE::rep>::digits >= BITS, "chrono::" #TYPE "::rep is not at least " #BITS " bits.");         \
+    static_assert(is_same_v<TYPE::period, __VA_ARGS__>, "chrono::" #TYPE "::period is not " #__VA_ARGS__ ".");
+
+DURATION_TEST(days, 25, ratio_multiply<ratio<24>, hours::period>)
+DURATION_TEST(weeks, 22, ratio_multiply<ratio<7>, days::period>)
+DURATION_TEST(months, 20, ratio_divide<years::period, ratio<12>>)
+DURATION_TEST(years, 17, ratio_multiply<ratio<146097, 400>, days::period>)
+
+// clang-format off
+static_assert(is_same_v<sys_seconds, time_point<system_clock, seconds>>,
+    "sys_seconds is not time_point<system_clock, seconds>.");
+static_assert(is_same_v<sys_days, time_point<system_clock, days>>,
+    "sys_days is not time_point<system_clock, days>.");
+
+static_assert( is_same_v<local_seconds, time_point<local_t, seconds>>,

```

Comment:
Super nitpick: this space (in a `clang-format off` region) doesn't appear to be aligning anything:
```suggestion
static_assert(is_same_v<local_seconds, time_point<local_t, seconds>>,
```